controllers/DataSetAdmin.js

```diff
@@ -0,0 +1,271 @@
+import {AdminController} from 'nxus-admin'
+import {templater} from 'nxus-templater'
+import {actions} from 'nxus-web'
+import {router} from 'nxus-router'
+import _ from 'underscore'
+import morph from 'morph'
+import nodeUUID from 'node-uuid'
+import {dataManager} from 'nxus-data-manager'
+
+import * as field from '../fieldConstants'
+
+export default class DataSetAdmin extends AdminController {
+
+  constructor() {
+    super({
+      modelIdentity: 'datasets-dataset',
+      icon: 'fa fa-table',
+      ignoreFields: ['id', 'fields', 'rowCount', 'createdAt', 'updatedAt' ],
+      instanceTitleField: 'DataSet'
+    })
+    //custom datasets edit form template
+    templater.replace().template(__dirname+"/../templates/datasets-dataset-form.ejs", this.pageTemplate, this.templatePrefix+"-edit")
+    //action and route registered for uploading associated DataRow's for a DataSet
+    actions.add(this.templatePrefix + "-list", "Upload", "/upload-data/", {
+      icon: "fa fa-upload",
+      group: "instance"
+    })
+    templater.replace().template(__dirname+"/../templates/datasets-dataset-admin-upload-datarow.ejs", this.pageTemplate, this.templatePrefix+"-upload-datarow")
+    router.route("GET", this.routePrefix+"/upload-data/:id", ::this.dataSetUploadDataRow)
+    router.route("POST", this.routePrefix+"/upload-data/:id/save", ::this.dataSetUploadDataRowSave)
+    //TODO - adjust when the data-manager is ready for use
+    dataManager.uploadPath(this.routePrefix+"/upload-data/:id/save", 'csvfile')
+  }
+
+  edit(req, res, query) {
+    return super.edit(req, res, query)
+    .then( (editContext) => {
+      let instance = editContext.object
+      this.log.debug( "editing dataset id:  ", instance.id)
+      return [this.models['datasets-datarow'].find({dataset: instance.id}), editContext]
+    }).spread( (datarows, context) => {
+      this.log.debug( "find returned matched datarows:  ", datarows)

```

Comment:
Question about some of these debug outputs, my feeling is that general debug lines should not dump full objects in most cases (not just be an equivalent of all the places you needed a console.log) - should still be reasonable to run with debug=true and use the log output. So are all these debugs generally useful?